### PR TITLE
Add fix numdiff based tests for bonded interactions

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

The fix numdiff command allows to tests consistency between energy and force computation via numerical differentiation of the energy. This adds a new test section to the test tools for bonded interactions to use these tests when the fix is available (i.e. the EXTRA-FIX package is installed).

**Related Issue(s)**

N/A

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

Several styles currently fail the test. Linear tabulation seems to have too large an error for this kind of test. For some the epsilon is adjusted. Others currently have the test disabled pending further investigation of the cause.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.

### CODE REVIEW

File: doc/src/Build_development.rst [hunk 1]
```diff
@@ -234,12 +241,14 @@ will be skipped if prerequisite features are not available in LAMMPS.
    time.  Preference is given to parts of the code base that are easy to
    test or commonly used.
 
-Tests for styles of the same kind of style (e.g. pair styles or bond
-styles) are performed with the same test executable using different
-input files in YAML format.  So to add a test for another style of the
-same kind it may be sufficient to add a suitable YAML file.
-:doc:`Detailed instructions for adding tests <Developer_unittest>` are
-provided in the Programmer Guide part of the manual.
+Many tests have special test programs that test individual functions or
```
Comment (stanmoore1) [inline doc/src/Build_development.rst]: This paragraph is a little confusing to me. Maybe it could be reworded?

**Review by stanmoore1 — APPROVED** [review]
Overall I approve, just had one comment

File: doc/src/Build_development.rst (hunk 1 above)
Comment (akohlmey) [reply doc/src/Build_development.rst]: I made some updates to this paragraph by describing everything with more details and thus be more specific. Is this what you are looking for?

Comment (stanmoore1) [reply doc/src/Build_development.rst]: Yes looks good, thanks. Just one typo: 
These special test programs are structured to ~have~ perform multiple individual tests

Comment (stanmoore1) [reply doc/src/Build_development.rst]: I approve

**Pull request merged**